cmd2/ansi.py

```diff
@@ -27,11 +27,47 @@
 ANSI_STYLE_RE = re.compile(r'\x1b\[[^m]*m')
 
 
+class ColorBase(Enum):
```

Comment:
I like how you extracted the common functionality to an Enum base class and also added overrides for the `+` operator.